compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1420,8 +1420,13 @@ public BType readType(int cpI) throws IOException {
                     return bMapType;
                 case TypeTags.INVOKABLE:
                     BInvokableType bInvokableType = new BInvokableType(null, null, null, null);
+                    bInvokableType.tsymbol = Symbols.createInvokableTypeSymbol(SymTag.FUNCTION_TYPE, flags,
```

Comment:
In the `setTSymbolForInvokableType`(L1437) method, no need to create a new type symbol again. We can remove that.

Reply:
Fixed 8795ec02ca7e346039059139ce2b9f8364ba7c29

Reply:
We now pass a type to `createInvokableTypeSymbol` though? May be OK, but it could also have side effects. For now, shall we leave as is and move the symbol creation here to within the check for any function?

Reply:
What do you mean by side effects? Isn't it redundant to have symbol creation in two places?

Reply:
Yeah, I understand your question. However, even if we move within the block, there may still be issues if there are places that generate a `function type`(function type means any function type) without a type symbol(As we generate a type symbol for every `function type` from the BIR). In order to fix this properly, shouldn't we move the flag that checks for the presence of a type symbol to the beginning?(What I mean is that we only need to create a type symbol from the BIR if it is present when writing to the BIR)

Reply:
> However, even if we move within the block, there may still be issues if there are places that generate a function type(function type means any function type) without a type symbol(As we generate a type symbol for every function type from the BIR).

What kind of issue? Shouldn't every `function` type have a symbol?

> In order to fix this properly, shouldn't we move the flag that checks for the presence of a type symbol to the beginning?(What I mean is that we only need to create a type symbol from the BIR if it is present when writing to the BIR) 

Yes, but we can't do this atm because it is not how `function` was initially handled, right? IMO, it's OK to create a symbol for `function` like this since the attributes are fixed.

Reply:
Every type should have a type symbol. 
After changing the logic as you mentioned, when reading function types from BIR, every `any function` type has a type symbol. However, in the normal compiling flow, there can be `any function` types without type symbols. my concern is whether this can cause any issues with the current compilation logic.

Reply:
What's the scenario where a `function` may not have a type symbol? How would adding the symbol here cause an issue?

Reply:
> For example, there may be places that check if tsymbol.type is null and do something based on that. Previously it was null, but now you're setting the type.
> 
Previously it was null only for `any function` types. For other function types, we create the `tsymbol` using the invokable-type

> I don't think it's "redundant" if you move it to within the if block after checking for `function`.
I mean if we do so, we will be creating a tsymbol in `setTSymbolForInvokableType(bInvokableType, bInvokableType.retType);` and witihin the if-block, so wouldn't it be good if we have this above the if-block?

Reply:
> > For example, there may be places that check if tsymbol.type is null and do something based on that. Previously it was null, but now you're setting the type.
> 
> Previously it was null only for `any function` types. For other function types, we create the `tsymbol` using the invokable-type

Is this correct? `null` is passed for `type` in https://github.com/ballerina-platform/ballerina-lang/pull/39934/files#diff-75a4933bf7e446d096c3a9ce1c8e8abb41cd5497adba2c1a9c84093240169519L1157?

> 
> > I don't think it's "redundant" if you move it to within the if block after checking for `function`.
> 
> I mean if we do so, we will be creating a tsymbol in `setTSymbolForInvokableType(bInvokableType, bInvokableType.retType);` and witihin the if-block, so wouldn't it be good if we have this above the if-block?

It's only created once because we return within the if block? It's either within the if block (for `function`) or in `setTSymbolForInvokableType` for other functions?

Reply:
Got it. Previously we haven't passed the type when creating the tsymbol.

Reply:
I think passing `null` wouldn't break any flows. 73c243508dd413bbb93623cfed4ffa886c017dc3